src/core/stream_recv.c

```diff
@@ -966,13 +974,25 @@ QuicStreamRecvFlush(
 
         QUIC_STATUS Status = QuicStreamIndicateEvent(Stream, &Event);
 
-        Stream->Flags.ReceiveCallActive = FALSE;
+        //
+        // To clear the top bit of RecvCompletionLength (the active receive call flag),
+        // we cannot subtract QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG
+        // because of overflow warning. Instead, we need to calcurate the difference between
+        // the actual number of complete bytes and RecvCompletionLength.
+        //
+        RecvCompletionLength = Stream->RecvCompletionLength;
+        uint64_t NumCompletedBytes = RecvCompletionLength & ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG;
+        RecvCompletionLength = InterlockedExchangeAdd64(
+            (int64_t*)&Stream->RecvCompletionLength,
+            NumCompletedBytes - RecvCompletionLength);
+        RecvCompletionLength &= ~QUIC_STREAM_RECV_COMPLETION_LENGTH_RECEIVE_CALL_ACTIVE_FLAG;
 
         if (Status == QUIC_STATUS_CONTINUE) {
             CXPLAT_DBG_ASSERT(!Stream->Flags.SentStopSending);
             InterlockedExchangeAdd64(
```

Comment:
It seems we could avoid this interlocked operation (and the other one below) since we always add here only to subtract right after.

We could basically have `BufferLength = RecvCompletionLength + Event.RECEIVE.TotalBufferLength` at line 1028, and when we substract from `Stream->RecvCompletionLength`, we can substract only `RecvCompletionLength`.